c_src/TLH_absTab.c

```diff
@@ -0,0 +1,225 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <glib.h>
```

Comment:
@gnuvince Conversation outside of this PR have suggested using an adapted version of the optics hash table (https://github.com/RAttab/optics/blob/master/src/utils/htable.h) as it's pretty dirt simple and can easily be adapted to our needs here.